### DISCUSSION TITLE
Start simulation with at random initial ranking

### DISCUSSION THREAD

**Discussion by MartijnOW** [asreview/asreview#1215] [Q&A]
Hi ASReview community,

During my exploration of ASReview, I came across a few points that were not entirely clear to me. So, I would like to ask those in three separated discussions and I hope you can help me out! Just to explain the context: I want to perform a batch simulation (from the command prompt) with a number of runs, let's say 10. For that I used this code: asreview simulate-batch My_data_file.csv --state_file batchrun_test.h5 --init_seed 1 --seed 1 --n_run 10

In this example, I did not explicitly state: -e tfidf -m nb -q max -b double, as these should be the default settings.

When starting a new screening in the interface, it is possible to start of with a ranking at random. However, I read in the documentation that leaving out the prior_idx command makes ASReview chose prior knowledge at random and not an at random initial ranking (so a ranking without any prior knowledge). Is it possible to start of with an initial ranking at random? I tried using one article for both relevant and irrelevant prior knowledge and it do get different results. However, I am not sure if it is working as I intended, so I hope you can help me out.

**Comment by Rensvandeschoot** [comment]
You can always choose `random` as a query strategy (see the [docs](https://asreview.readthedocs.io/en/latest/reference.html#module-asreview.models.query) for more options) and even if you select prior knowledge the record shown in the reviewing screen is chosen randomly. You can also design your own query strategy using a [template](https://github.com/asreview/template-extension-new-model) to design a query strategy exactly as you want it.

**Reply by MartijnOW** [reply]
Thanks Rens, just one quick follow up question: do you know if this is still the appropriate command within the makita tool? I got it to work in the older version, but within makita I keep getting errors.

**Reply by Rensvandeschoot** [reply]
Could you share the errors? (ping: @JTeijema )

**Reply by MartijnOW** [reply]
Of course! So when I try to adapt the query, I used this explanation: https://asreview.readthedocs.io/en/stable/cli.html#query-strategies-table. I tried both -q and --query_strategy, but for both I got the error below:

- asreview makita: error: unrecognized arguments: -q random
- asreview makita: error: unrecognized arguments: --query_strategy random

Do I do anything wrong here?

Entire command is: 
asreview makita template basic -f jobs_DE.bat --init_seed 1 --model_seed 1 --n_runs 10 --query_strategy random

**Reply by jteijema** [reply]
Hi @MartijnOW,

Setting the query strategy isn't currently implemented into Makita. However, you can manually edit the jobs file to add it to the simulation line.

Moreover, you could also edit a template to fit your own needs, such as a random query strategy. Instructions for your own template can be found on the [repository](https://github.com/asreview/asreview-makita#create-and-use-custom-templates).

For your case, I edited the `basic` template and added a query strategy to the line with the simulation command, following the steps. Example output plot: 

![image](https://user-images.githubusercontent.com/28191416/194562671-dc05e56e-3f68-4c0d-8792-182e5da02d8a.png)

Hope this helps!
Jelle

**Reply by MartijnOW** [reply]
Thanks @JTeijema! That helps a lot. However, I do not see any change in the jobs file yet. Is this changed after the next update? And is it perhaps possible to schedule a short call. I have a few other minor issues I experienced with ASReview and I think a short call may perhaps be most efficient :)

**Reply by Rensvandeschoot** [reply]
Dear @MartijnOW,

The query strategy can be set in the code to be run in the CLI, just like many other options you can specify; see the overview of options in the [documentation](https://asreview.readthedocs.io/en/latest/simulation_cli.html#command-line-arguments-for-simulating). [Makita ](https://github.com/asreview/asreview-makita) is 'just' an easy way to generate a batch file containing many lines of code without the need to write every line yourself. 

So, Makita is beneficial if you want to run the same script for many different datasets  (see, for example, the SH-files used in [this ](https://github.com/asreview/paper-guidelines-KIFMS)study) or if you want to compare many different models (see, for example, the SH-files in [this ](https://github.com/JTeijema/Mega-Meta-paper-simulations) project). We have provided some templates you can use. 

Suppose you want to generate a batch script with different code than we provide in the example. In that case, (e.g., using the query strategy's random option), you should write your own template file (which you can donate to the project so that other users can use your contribution). 

If you want to run a single simulation study, it might be easier to follow the instructions about running a simulation study in the [web-app ](https://asreview.readthedocs.io/en/latest/simulation_webapp.html)or via the [CLI](https://asreview.readthedocs.io/en/latest/simulation_cli.html#simulation-via-command-line). 

Hope this helped!

**Reply by MartijnOW** [reply]
Thank you Rens. I understand that makita is 'just' a tool to make things easier and thank you for the suggestion of creating/adapting a template. My apologies that I keep returning to this thread, but I keep running into some issues. Unfortunately, it is not possible to perform single simulations, as I would love to eventually perform 1,000 simulations to assess variation. I have tried a couple of things to get a new random order per simulation and I have edited the commands under :: Simulate runs jobs.bat file to the following: 

jobs.bat file 

:: Simulate runs
mkdir output\simulation\HEA_Early_Detection_CVD_2016_2022_DE\state_files
asreview simulate data\data.xlsx -s output\simulation\data\state_files\sim_data_0.asreview -q random --init_seed 1 --seed 1 
asreview metrics output\simulation\data\state_files\sim_data_0.asreview -o output\simulation\data\metrics_sim_data_0.json
asreview simulate data\data.xlsx -s output\simulation\data\state_files\sim_data_1.asreview -q random --init_seed 2 --seed 2
asreview metrics output\simulation\data\state_files\sim_data_1.asreview -o output\simulation\data\metrics_sim_data_1.json
asreview simulate data\data.xlsx -s output\simulation\data\state_files\sim_data_2.asreview -q random --init_seed 3 --seed 3 
asreview metrics output\simulation\data\state_files\sim_data_2.asreview -o output\simulation\data\metrics_sim_data_2.json

This seems to indeed lead to a ranking without any prior knowledge. However, the issue is that I cannot make the random ranking vary per simulation. When changing the init_seed the outcomes remain the same and the same goes for changing the model seed or doing both. Is there a way to get a different random ranking per simulation?

**Reply by Rensvandeschoot** [reply]
@MartijnOW I created an issue for your questions because there might be something wrong with the seeding functionality. This is further discussed at https://github.com/asreview/asreview/issues/1264. 
Could you add an example dataset for reproducing the issue in this new thread?

**Comment by Rensvandeschoot** [comment]
From a user's perspective, why would you want to start randomly? It will take you a lot of time to find the first relevant record. Most often a user knows some papers that should be found in a systematic review, see also the blog post describing [seven ways to integrate ASReview in your workflow](https://asreview.nl/blog/seven-ways-to-integrate-asreview/).

**Reply by MartijnOW** [reply]
Thank you Rens! That is helping me a lot. 

And to answer your second question: I would like to know the variation in 'time to find all relevant articles'. I would like to do this with random prior knowledge (letting ASReview choose prior knowledge at random, with a seed of course) and also compare it to no prior knowledge. I've not performed the analyses yet, but I can imagine that variation will be much larger without prior knowledge as time to first relevant article may vary a lot. On the other hand, you do feed ASReview with (negative) information, thereby accelerating the process from the point when you find the first relevant article and thus limiting the variation.

I agree that it is likely a reviewer already has some information, but it is also good to know whether it is 'necessary' to have prior knowledge and how large the impact of prior knowledge is, both on variation and average time to find all relevant articles

**Reply by Rensvandeschoot** [reply]
This is a very nice research question!! Will you share your results during our next Users Meeting?? (No date yet)

**Reply by MartijnOW** [reply]
Sure! Let me know when you have a date for the Users meeting

**Reply by Rensvandeschoot** [reply]
@MartijnOW, we are planning our next users meeting in March, are you interested in presenting your findings? I would also like to cite your study, do you have a reference?
